EDU-3964: Update docs and samples for Nexus Go GA (#3334)
temporalio/documentation: docs/develop/go/temporal-nexus.mdx

Nexus Operation handlers are typically defined in the same Worker as the underlying Temporal primitives they abstract.
Operation handlers can decide if a given Nexus Operation will be synchronous or asynchronous, execute arbitrary code, and invoke underlying Temporal primitives such as a Workflow, Query, Signal, or Update.

The `temporalnexus` package has builders to create Nexus Operations and other helpers for authoring Operation handlers:

- `NewWorkflowRunOperation` \- Run a Workflow as an asynchronous Nexus Operation
- `GetClient` \- Get the Temporal Client that the Worker was initialized with for synchronous RPC handlers backed by
  Temporal primitives such as for Signals, and Queries

> This tutorial starts with a sync Operation handler example using the `nexus.NewSyncOperation` method, and then uses `NewWorkflowRunOperation` to start a handler Workflow from a Nexus Operation.

Reviewer: ```suggestion
This tutorial starts with a sync Operation handler example using the `nexus.NewSyncOperation` method, and then shows how to create an async Operation handler that uses `NewWorkflowRunOperation` to start a handler Workflow from a Nexus Operation.
```

### Develop a Synchronous Nexus Operation handler

The `nexus.NewSyncOperation` builder function is for exposing simple RPC handlers.

> Its handler function is provided with an SDK client that can be used for signaling, querying, and listing Workflows.

Reviewer 2: ```suggestion
```
This is no longer true

Nexus Operation handlers are typically defined in the same Worker as the underlying Temporal primitives they abstract.
Operation handlers can decide if a given Nexus Operation will be synchronous or asynchronous, execute arbitrary code, and invoke underlying Temporal primitives such as a Workflow, Query, Signal, or Update.

The `temporalnexus` package has builders to create Nexus Operations and other helpers for authoring Operation handlers:

- `NewWorkflowRunOperation` \- Run a Workflow as an asynchronous Nexus Operation
- `GetClient` \- Get the Temporal Client that the Worker was initialized with for synchronous RPC handlers backed by
  Temporal primitives such as for Signals, and Queries

This tutorial starts with a sync Operation handler example using the `nexus.NewSyncOperation` method, and then shows how to create an async Operation handler that uses `NewWorkflowRunOperation` to start a handler Workflow from a Nexus Operation.

### Develop a Synchronous Nexus Operation handler

The `nexus.NewSyncOperation` builder function is for exposing simple RPC handlers.

> Typically to use SDK client, which is obtained via `temporalnexus.GetClient(ctx)`, for signaling, querying, and listing Workflows.

Reviewer 3: ```suggestion
Typically this is used to access the SDK client, which is obtained via `temporalnexus.GetClient(ctx)`, for signaling, querying, and listing Workflows.
```
